enhancements/api-review/add-unqualified-search-registries.md

```diff
@@ -0,0 +1,211 @@
+---
+title: add-unqualified-search-registries
+authors:
+  - "@umohnani8"
+reviewers:
+  - "@mrunalp"
+  - "@derekwaynecarr"
+approvers:
+  - "@mrunalp"
+  - "@derekwaynecarr"
+creation-date: 2020-06-16
+last-updated: 2020-06-16
+status: implementable
+---
+
+# Add UnqualifiedSearchRegistries to cluster wide Image CR
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+
+## Summary
+
+Today, the cluster wide Image CR (images.config.openshift.io cluster) doesn't have a way of setting the
+registries to be used when pulling an image using short name.
+
+With the move the v2 /etc/containers/registries.conf, there is a new
+option called `unqualified-search-registries`. This is the list of registries that the container tools will check when pulling an image using its short name.
+
+## Motivation
+
+Today, when a user wants to pull images using short names, they have to create two `machineConfigs` (one for the master pool and one for the worker pool) that create a file under `/etc/containers/registries.conf.d` with the changes they want for `unqualified-search-registries`. 
+
+Users may have multiple internal registries that they use for pulling images. The DNS of these registries could change, and that would require changing the image spec to match this, which can be tedious. Using `unqualified-search-registries` to be able to configure the list of registries to search from allows the user to use short names for image pulls.
+
+Users can currently configure `AllowedRegistries`, `BlockedRegistries`, and `InsecureRegistries` in the cluster wide Image CR. Adding the `unqualified-search-registries` option to this cluster wide Image CR will make short name configuration for users simple and have all the image related configurations in one CR.
+
+
+### Goals
+
+- Enable the user to be able to set `unqualified-search-registries` in the cluster wide Image CR.
+
+### Non-Goals
+
+## Proposal
+
+The Image API is extended by adding an optional `UnqualifiedSearchRegistries` field with type `[]string` to `RegistrySources`:
+
+```go
+// RegistrySources holds cluster-wide information about how to handle the registries config.
+type RegistrySources struct {
+    // ...
+    
+	// unqualifiedSearchRegistries are registries that will be searched when pulling images using short names.
+	// +optional
+	UnqualifiedSearchRegistries []string `json:"unqualifiedSearchRegistries,omitempty"`
+}
+```
+
+The containerRuntimeConfig controller in the MCO already watches the cluster wide images.config.openshift.io CR for the allowed, blocked, and insecure registries. It will now watch for `unqualified-search-registries` and update `/etc/containers/registries.conf` accordingly.
```

Comment:
now watch for `unqualified-search-registries` as well and update

Reply:
Fixed